sql/core/src/main/scala/org/apache/spark/sql/execution/python/EvaluatePython.scala

```diff
@@ -120,10 +121,18 @@ object EvaluatePython {
     case (c: java.math.BigDecimal, dt: DecimalType) => Decimal(c, dt.precision, dt.scale)
 
     case (c: Int, DateType) => c
+    // Pyrolite will unpickle a Python datetime.date to a java.util.Calendar
+    case (c: Calendar, DateType) => DateTimeUtils.fromJavaCalendarForDate(c)
```

Comment:
so we will never hit this?

Reply:
seems what we need is a `case (c: Calendar, StringType) => ...`

Reply:
I think he did this in Python side because here we don't know if `Calendar` is from `datetime.date` or `datetime.datetime`.